lib/cache/collections.go

```diff
@@ -1041,10 +1041,15 @@ func (c *clusterConfig) fetch(ctx context.Context) (apply func(ctx context.Conte
 			return nil
 		}
 		c.setTTL(clusterConfig)
+
+		// To ensure backward compatibility, ClusterConfig resources/events may
+		// feature fields that now belong to separate resources/events. Since this
+		// code is able to process the new events, ignore any such legacy fields.
+		// DELETE IN 8.0.0
+		clusterConfig.ClearLegacyFields()
```

Comment:
I don't understand why we need to clear the legacy fields at the cache layer
what happens if we keep them?

Reply:
As you can see this is immediately followed by a `SetClusterConfig` call. If the legacy fields were left intact the following conditions would trigger:
https://github.com/gravitational/teleport/blob/6ce105608f296a797fcb493acf8df32c384e41e7/lib/services/local/configuration.go#L267-L273